lib/ansiblelint/__main__.py

```diff
@@ -97,20 +98,21 @@ def report_outcome(matches: List["MatchError"], options) -> int:
 warn_list:  # or 'skip_list' to silence them completely
 """
 
-    matched_rules = {match.rule.id: match.rule.shortdesc for match in matches}
-    for id in sorted(matched_rules.keys()):
-        if id not in options.warn_list:
-            msg += f"  - '{id}'  # {matched_rules[id]}\n"
+    matched_rules = OrderedDict({match.rule.id: match.rule for match in matches})
```

Comment:
There's no point in making an unordered dict unless you keep it ordered originally.

Reply:
Also, intermediate in-memory dict is unnecessary

Reply:
```suggestion
    matched_rules = OrderedDict((match.rule.id, match.rule) for match in matches)
```

Reply:
P.S. Starting Python 3.6 dicts are ordered in CPython (de-facto) and with 3.7+ they are ordered de-jure. So there's probably no point in importing `OrderedDict`.